aptos-move/framework/aptos-framework/sources/fungible_asset.move

```diff
@@ -437,8 +442,13 @@ module aptos_framework::fungible_asset {
     /// Check whether the balance of a given store is >= `amount`.
     public fun is_balance_at_least<T: key>(store: Object<T>, amount: u64): bool acquires FungibleStore {
         let store_addr = object::object_address(&store);
-        if (store_exists(store_addr)) {
-            borrow_store_resource(&store).balance >= amount
+        is_address_balance_at_least(store_addr, amount)
+    }
+
+    /// Check whether the balance of a given store is >= `amount`.
+    public(friend) fun is_address_balance_at_least(store_addr: address, amount: u64): bool acquires FungibleStore {
```

Comment:
Why not use is_balance_at_least? Why do we need this friend-only version?

Reply:
removing reading ObjectCore improves performance by more than 10% (i.e. reduces regression by 10%)

we could theoretically have some private function in object to give us Object<T> without checking for T and ObjectCore, but that seems worse to me (as than the invariants stop holding)

Reply:
Gotcha sounds good then

Reply:
I reminds me when I started with those api with address only.